distributed/client.py

```diff
@@ -251,9 +251,10 @@ def type(self):
         return self._state.type
 
     def release(self):
-        if not self._cleared and self.client.generation == self._generation:
-            self._cleared = True
-            self.client._dec_ref(tokey(self.key))
+        with self.client._lock:
+            if not self._cleared and self.client.generation == self._generation:
+                self._cleared = True
+                self.client._dec_ref(tokey(self.key))
```

Comment:
What is the likelihood that a decref operation triggers other lock-taking operations (e.g. by way of garbage collection)?